client/rest-high-level/src/main/java/org/elasticsearch/client/tasks/ElasticsearchException.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.tasks;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.XContentParser;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+/**
+ * client side counterpart of server side
+ * {@link org.elasticsearch.ElasticsearchException}
+ */
+public class ElasticsearchException  extends RuntimeException  {
```

Comment:
as per our discussion we dont need this class at all.

Reply:
Some thoughts:

I checked the `org.elasticsearch.action.support.tasks.BaseTasksResponse` and I think that full `ElasticSeachException` are marshalled on the wire.
Also I agree with you, the user does not need a full blown Exception on the client side.
That said, what happens in the marhsalling is that it is basically a tree of exception-causes, where the cause is another `ElasticSearchException` (with suppressed in between).
So what if I did a "shallow" exception, that does not inherit from `RuntimeException` but it is self consistent? 
Maybe some user is interested in what has caused an exception and more detailed information . This way I would save the information content for the user without compromising parsing.